stdlib/3/unittest/mock.pyi

```diff
@@ -164,35 +164,63 @@ class _patch(Generic[_T]):
     autospec: Any
     kwargs: Mapping[str, Any]
     additional_patchers: Any
-    @overload
-    def __init__(
-        self: _patch[MagicMock],
-        getter: Callable[[], Any],
-        attribute: str,
-        *,
-        spec: Optional[Any],
-        create: bool,
-        spec_set: Optional[Any],
-        autospec: Optional[Any],
-        new_callable: Optional[Any],
-        kwargs: Mapping[str, Any],
-    ) -> None: ...
-    # This overload also covers the case, where new==DEFAULT. In this case, self is _patch[Any].
-    # Ideally we'd be able to add an overload for it so that self is _patch[MagicMock],
-    # but that's impossible with the current type system.
-    @overload
-    def __init__(
-        self: _patch[_T],
-        getter: Callable[[], Any],
-        attribute: str,
-        new: _T,
-        spec: Optional[Any],
-        create: bool,
-        spec_set: Optional[Any],
-        autospec: Optional[Any],
-        new_callable: Optional[Any],
-        kwargs: Mapping[str, Any],
-    ) -> None: ...
+    if sys.version_info >= (3, 8):
+        @overload
+        def __init__(
+            self: _patch[Union[MagicMock, AsyncMock]],
```

Comment:
Looking over it again, this overload seems wrong to me, even before the PR. This defines the default type of a patch, if `new` is not specified. This can never happen as it is a required argument. We should probably just get rid of all overloads and version checks for the `_patch.__init__()` constructor and just keep the second overload (where `new: _T`).

If you want to, you can do that, otherwise I'd do it in a separate PR.

Reply:
@srittau Thanks for the feedback. I think I know what you mean. However, since I'm relatively new to typing stubs it might be faster if you do it in a separate PR instead of multiple back and forth if that's ok with you.